### PR TITLE
🚧 HNXBJ4 task: Fix context search JSONL row freshness [202605141005-HNXBJ4]

### PR DESCRIPTION
**Pull request by densmirnov**
Task: `202605141005-HNXBJ4`
Title: Fix context search JSONL row freshness
Canonical task record: `.agentplane/tasks/202605141005-HNXBJ4/README.md`

## Summary

Fix context search JSONL row freshness

Context search marks fresh JSONL facts and graph projection rows as stale because row-level projection hashes are compared to whole-file hashes. Fix row-level freshness and cover it with regression tests so empty-folder semantic assimilation search reports fresh derived rows.

## Scope

- In scope: Context search marks fresh JSONL facts and graph projection rows as stale because row-level projection hashes are compared to whole-file hashes. Fix row-level freshness and cover it with regression tests so empty-folder semantic assimilation search reports fresh derived rows.
- Out of scope: unrelated refactors not required for "Fix context search JSONL row freshness".

## Verification

- State: ok
- Note:

```text
Fixed context search JSONL row freshness by recomputing current row hashes for row-level projection
refs. Verified focused context tests, eslint/prettier, release/docs gates, platform-critical, and
empty-folder semantic assimilation smoke with no stale search rows.
```
- Canonical workflow state lives in the task README.

<details>
<summary>Raw evidence</summary>

- Updated: 2026-05-14T10:10:38.032Z
- Branch: task/202605141005-HNXBJ4/context-jsonl-freshness
- Head: b9a6042a9ccc

```text
 .../src/commands/context/release-readiness.test.ts | 55 ++++++++++++++++++++++
 packages/agentplane/src/commands/context/search.ts | 33 +++++++++++++
 2 files changed, 88 insertions(+)
```

</details>

### CODE REVIEW

File: packages/agentplane/src/commands/context/search.ts [hunk 1]
```diff
@@ -188,6 +191,36 @@ async function buildFreshness(
...
+      projection_sha256: projectionSha256,
+      file_sha256: fileSha256,
+      stale: fileSha256 !== projectionSha256,
+    };
+  }
+  const expectedId = marker.slice(separator + 1);
+  const rows = parseJsonlLines(await readText(filePath));
+  for (const [index, row] of rows.entries()) {
+    const id = typeof row.id === "string" ? row.id : String(index + 1);
+    if (id !== expectedId) continue;
```
[the remark this answers is not in the document]
Comment (densmirnov) [reply packages/agentplane/src/commands/context/search.ts]: Fixed in cd64a7fbf: row id normalization now matches reindex (`String(row.id ?? index + 1)`) for both live JSONL search refs and projection freshness lookup. Added regression coverage for numeric JSONL ids (`facts.jsonl#fact=42`) and reran focused context tests, platform-critical, release:check, and empty-folder semantic assimilation smoke.
